src/IO/CompressionMethod.h

```diff
@@ -47,6 +47,14 @@ std::string toContentEncodingName(CompressionMethod method);
   */
 CompressionMethod chooseCompressionMethod(const std::string & path, const std::string & hint);
 
+std::unique_ptr<ReadBuffer> wrapReadBufferWithCompressionMethod(
+    std::unique_ptr<ReadBuffer> nested,
+    CompressionMethod method,
+    const Settings &settings,
```

Comment:
I still think that you don't need to pass `Settings` here, `settings.zstd_window_log_max` is enough to pass here directly.